lib/tasks/hosts.rake

```diff
@@ -34,14 +34,15 @@ namespace :hosts do
     missingdns = []
     offline = []
 
-    Host.out_of_sync(1.hour.ago).all(:order => 'environment_id asc').collect do |host|
+    Host::Managed.out_of_sync.order('environment_id ASC').collect do |host|
       $stdout.flush
       ip = Resolv::DNS.new.getaddress(host.name).to_s rescue nil
       if ip.empty?
         missingdns << host
       else
         puts "conflict IP address for #{host.name}" unless ip == host.ip
-        if Ping.pingecho host.ip
+        host_icmp = Net::Ping::TCP.new(host.ip)
```

Comment:
if you are just using tcp ping, please change: 

1. require line to ```require 'net/ping/tcp'```
2. the variable name is wrong
3. ECONNREFUSED error is considered a failed ping, yet in our case it shows the machine is up (see https://github.com/chernesk/net-ping/blob/master/doc/ping.txt)

TBH: maybe using Net::Ping::External will yield more reliable results.